R/reporter-debug.R

```diff
@@ -0,0 +1,51 @@
+#' @include reporter.R
+NULL
+
+#' Test reporter: start recovery.
+#'
+#' This reporter will call a modified version of \code{\link{recover}} on all
+#' broken expectations.
+#'
+#' @export
+DebugReporter <- R6::R6Class("DebugReporter", inherit = Reporter,
+  public = list(
+    add_result = function(context, test, result) {
+      if (!expectation_success(result)) {
+        if (sink.number() > 0) {
+          sink(self$out)
+          on.exit(sink(), add = TRUE)
+        }
+
+        recover(start_frame = result$start_frame,
+                end_frame = result$end_frame)
+      }
+    }
+  )
+)
+
+# Copied from utils::recover()
```

Comment:
Can you please also copy the copyright statement?  It might be better to move into completely a new file. Will also need to update `Authors@R` to reference R core